src/Compilers/VisualBasic/Test/Symbol/SymbolsTests/SymbolErrorTests.vb

```diff
@@ -6118,6 +6118,9 @@ End Class
 BC30573: Error in project-level import '<xmlns:p="p2">' at '<xmlns:p="p2">' : XML namespace prefix 'p' is already declared.
 BC30573: Error in project-level import '<xmlns="default2">' at '<xmlns="default2">' : XML namespace prefix '' is already declared.
 ]]></errors>)
+            Dim embedded = compilation.GetTypeByMetadataName("Microsoft.VisualBasic.Embedded")
+            Assert.IsType(Of EmbeddedSymbolManager.EmbeddedNamedTypeSymbol)(embedded)
```

Comment:
From discussion with Aleksey, there may be a way to get a serializable `EmbeddedNamedTypeSymbol` by using a partial. I'll try that.

Reply:
@AlekseyTs I tried the "partial" approach we discussed, but it didn't work.
I didn't trace all the way to the root cause, but there are assertions in `EmbbedNamedTypeSymbol` that it only has a single syntax reference. So I don't believe it will be possible to add the `Serializable` attribute to an existing embedded type.

```VB
        Friend NotInheritable Class EmbeddedNamedTypeSymbol
       
            Public Sub New(...)
                ...
#If DEBUG Then
                Dim references As ImmutableArray(Of SyntaxReference) = decl.SyntaxReferences
                Debug.Assert(references.Length() = 1)
                Debug.Assert(references.First.SyntaxTree.IsEmbeddedSyntaxTree())
#End If
                ...
            End Sub
```

---
In reply to: [164609829](https://github.com/dotnet/roslyn/pull/24336#discussion_r164609829) [](ancestors = 164609829)